include/ignition/rendering/RenderEngine.hh

```diff
@@ -163,6 +164,10 @@ namespace ignition
       public: virtual ScenePtr CreateScene(unsigned int _id,
                   const std::string &_name) = 0;
 
+      /// \brief Returns the GraphicsAPI currently in use
+      /// \return GraphicsAPI currently in use
+      public: virtual GraphicsAPI CurrentGraphicsAPI() const = 0;
```

Comment:
how about keeping the origin API function name, i.e. `public: virtual rendering::GraphicsAPI GraphicsAPI() = 0`?

Reply:
That was my first attempt, but it results in a hideous compiler. For some reason it believes the function name cannot be an enum (but it is fine if declared elsewhere. There may be a namespace problem or something)

Reply:
That was my first attempt, but it results in a hideous compiler. For some reason it believes the function name cannot be an enum (but it is fine if declared elsewhere. There may be a namespace problem or something)

Reply:
The patch below should make it work:

<details>
<summary>patch</summary>

```cpp
diff --git a/include/ignition/rendering/RenderEngine.hh b/include/ignition/rendering/RenderEngine.hh
index 00bd708d..cb478f4d 100644
--- a/include/ignition/rendering/RenderEngine.hh
+++ b/include/ignition/rendering/RenderEngine.hh
@@ -166,7 +166,7 @@ namespace ignition
 
       /// \brief Returns the GraphicsAPI currently in use
       /// \return GraphicsAPI currently in use
-      public: virtual GraphicsAPI CurrentGraphicsAPI() const = 0;
+      public: virtual rendering::GraphicsAPI GraphicsAPI() const = 0;
 
       /// \brief Set headless mode
       /// Only available in OGRE 2.2, which makes use of EGL
diff --git a/include/ignition/rendering/base/BaseRenderEngine.hh b/include/ignition/rendering/base/BaseRenderEngine.hh
index c84b65b7..cfb3a458 100644
--- a/include/ignition/rendering/base/BaseRenderEngine.hh
+++ b/include/ignition/rendering/base/BaseRenderEngine.hh
@@ -89,7 +89,7 @@ namespace ignition
       public: virtual void AddResourcePath(const std::string &_path) override;
 
       // Documentation Inherited
-      public: virtual GraphicsAPI CurrentGraphicsAPI() const override
+      public: virtual rendering::GraphicsAPI GraphicsAPI() const override
       {
         return GraphicsAPI::UNKNOWN;
       }
diff --git a/ogre/include/ignition/rendering/ogre/OgreRenderEngine.hh b/ogre/include/ignition/rendering/ogre/OgreRenderEngine.hh
index 70150e50..053a6364 100644
--- a/ogre/include/ignition/rendering/ogre/OgreRenderEngine.hh
+++ b/ogre/include/ignition/rendering/ogre/OgreRenderEngine.hh
@@ -101,7 +101,7 @@ namespace ignition
       public: void AddResourcePath(const std::string &_uri) override;
 
       // Documentation Inherited
-      public: virtual GraphicsAPI CurrentGraphicsAPI() const override;
+      public: virtual rendering::GraphicsAPI GraphicsAPI() const override;
 
       public: virtual Ogre::Root *OgreRoot() const;
 
diff --git a/ogre/src/OgreRenderEngine.cc b/ogre/src/OgreRenderEngine.cc
index 619b32f4..c3ddefd3 100644
--- a/ogre/src/OgreRenderEngine.cc
+++ b/ogre/src/OgreRenderEngine.cc
@@ -246,7 +246,7 @@ void OgreRenderEngine::AddResourcePath(const std::string &_uri)
 }
 
 //////////////////////////////////////////////////
-GraphicsAPI OgreRenderEngine::CurrentGraphicsAPI() const
+GraphicsAPI OgreRenderEngine::GraphicsAPI() const
 {
   return GraphicsAPI::OPENGL;
 }
diff --git a/ogre2/include/ignition/rendering/ogre2/Ogre2RenderEngine.hh b/ogre2/include/ignition/rendering/ogre2/Ogre2RenderEngine.hh
index f5ee5c6f..cb10b9e1 100644
--- a/ogre2/include/ignition/rendering/ogre2/Ogre2RenderEngine.hh
+++ b/ogre2/include/ignition/rendering/ogre2/Ogre2RenderEngine.hh
@@ -121,7 +121,7 @@ namespace ignition
                   const double _ratio, const unsigned int _antiAliasing);
 
       // Documentation Inherited
-      public: virtual GraphicsAPI CurrentGraphicsAPI() const override;
+      public: virtual rendering::GraphicsAPI GraphicsAPI() const override;
 
       /// \brief Create a scene
       /// \param[in] _id Unique scene Id
diff --git a/ogre2/src/Ogre2Material.cc b/ogre2/src/Ogre2Material.cc
index 2f1868f8..8e6ddbda 100644
--- a/ogre2/src/Ogre2Material.cc
+++ b/ogre2/src/Ogre2Material.cc
@@ -776,7 +776,7 @@ void Ogre2Material::UpdateShaderParams(ConstShaderParamsPtr _params,
                << name_param.first << std::endl;
         continue;
       }
-      if (Ogre2RenderEngine::Instance()->CurrentGraphicsAPI() ==
+      if (Ogre2RenderEngine::Instance()->GraphicsAPI() ==
           GraphicsAPI::OPENGL)
       {
         // set the texture map index
@@ -1119,7 +1119,7 @@ void Ogre2Material::SetVertexShader(const std::string &_path)
         "_ign_" + baseName,
         Ogre::ResourceGroupManager::DEFAULT_RESOURCE_GROUP_NAME,
         this->dataPtr->shaderLanguageCode(
-            Ogre2RenderEngine::Instance()->CurrentGraphicsAPI()),
+            Ogre2RenderEngine::Instance()->GraphicsAPI()),
         Ogre::GpuProgramType::GPT_VERTEX_PROGRAM);
 
   vertexShader->setSourceFile(_path);
@@ -1172,11 +1172,11 @@ void Ogre2Material::SetFragmentShader(const std::string &_path)
         "_ign_" + baseName,
         Ogre::ResourceGroupManager::DEFAULT_RESOURCE_GROUP_NAME,
         this->dataPtr->shaderLanguageCode(
-            Ogre2RenderEngine::Instance()->CurrentGraphicsAPI()),
+            Ogre2RenderEngine::Instance()->GraphicsAPI()),
         Ogre::GpuProgramType::GPT_FRAGMENT_PROGRAM);
 
   // set shader language specific parameters
-  if (Ogre2RenderEngine::Instance()->CurrentGraphicsAPI() == GraphicsAPI::METAL)
+  if (Ogre2RenderEngine::Instance()->GraphicsAPI() == GraphicsAPI::METAL)
   {
     // must set reflection pair hint for Metal fragment shaders
     // otherwise the parameters (uniforms) will not be set correctly
diff --git a/ogre2/src/Ogre2RenderEngine.cc b/ogre2/src/Ogre2RenderEngine.cc
index 8694c047..ab29fa57 100644
--- a/ogre2/src/Ogre2RenderEngine.cc
+++ b/ogre2/src/Ogre2RenderEngine.cc
@@ -1006,7 +1006,7 @@ std::string Ogre2RenderEngine::CreateRenderWindow(const std::string &_handle,
 }
 
 //////////////////////////////////////////////////
-GraphicsAPI Ogre2RenderEngine::CurrentGraphicsAPI() const
+GraphicsAPI Ogre2RenderEngine::GraphicsAPI() const
 {
   return this->dataPtr->graphicsAPI;
 }
```

</details>

Reply:
Done!